src/Core/Options.idr

```diff
@@ -229,11 +231,21 @@ defaultElab : ElabDirectives
 defaultElab = MkElabDirectives True True CoveringOnly 3 50 50 True
 
 export
-defaults : Options
-defaults = MkOptions defaultDirs defaultPPrint defaultSession
-                     defaultElab Nothing Nothing
-                     (MkPrimNs Nothing Nothing Nothing Nothing) []
-                     []
+defaultHashFn : Core String
+defaultHashFn
+    = do Nothing <- coreLift $ pathLookup ["sha256sum", "gsha256sum", "sha256"]
+           | Just p => pure p
+         Nothing <- coreLift $ pathLookup ["openssl"]
+           | Just p => pure "openssl sha256 -r"
+         pure "sha256sum"
```

Comment:
Should this be an exception? We've already attempted pathLookup on `"sha256sum"`
so a priori it does not exist and returning it has no chance of success?

Reply:
Yeah, it should be an exception.

Reply:
Amending

Reply:
Amended for an error explaining the attempts.